File: Pascal's-Triangle-II.cpp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<int> row(rowIndex + 1, 1);
        for (int i = 1; i <= rowIndex / 2; i++) {
            row[i] = (long long)row[i - 1] * (rowIndex - i + 1) / i;
            row[rowIndex - i] = row[i];
        }
        return row;
    }
};

File: Pascal's-Triangle-II.java
class Solution {
    public List<Integer> getRow(int rowIndex) {
        List<Integer> row = new ArrayList<>(Collections.nCopies(rowIndex + 1, 1));
        for (int i = 1; i <= rowIndex / 2; i++) {
            row.set(i, (int)((long)row.get(i - 1) * (rowIndex - i + 1) / i));
            row.set(rowIndex - i, row.get(i));
        }
        return row;
    }
}
